framework/src/outputs/FileOutput.C

```diff
@@ -74,12 +77,24 @@ FileOutput::FileOutput(const InputParameters & parameters) :
     _file_base += buffer;
   }
 
-  // Check the file directory of file_base
+  // Check the file directory of file_base and create if needed
   std::string base = "./" + _file_base;
   base = base.substr(0, base.find_last_of('/'));
   if (access(base.c_str(), W_OK) == -1)
-    mooseError("Can not write to directory: " + base + " for file base: " + _file_base);
-
+  {
+    //Directory does not exist. Loop through incremental directories and create as needed.
+    size_t pos = 0;
+    std::string inc_path;
+    do
```

Comment:
Yes, I know this would happen. It was just a weird of caution. And I'm
usually the one who's being told not to underestimate the breadth of the
capability spectrum of the users...

On Fri, May 27, 2016, 7:54 PM Derek Gaston <notifications@github.com> wrote:

> In framework/src/outputs/FileOutput.C
> <https://github.com/idaholab/moose/pull/7101#discussion_r64982221>:
>
> >    std::string base = "./" + _file_base;
> >    base = base.substr(0, base.find_last_of('/'));
> >    if (access(base.c_str(), W_OK) == -1)
> > -    mooseError("Can not write to directory: " + base + " for file base: " + _file_base);
> > -
> > +  {
> > +    //Directory does not exist. Loop through incremental directories and create as needed.
> > +    size_t pos = 0;
> > +    std::string inc_path;
> > +    do
>
> The same would happen with @bwspenc <https://github.com/bwspenc> 's code
> here.
>
> How about people just don't put / in their file or directory names?
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/idaholab/moose/pull/7101/files/bd2be34efac386b494aa5f1183bfd7561098d68d#r64982221>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAMWPuuKzyokcy9Acgub65kWx0-6aAEFks5qF6BrgaJpZM4IoKUf>
> .
>